cmd/fitgen/internal/profile/transform.go

```diff
@@ -162,7 +162,8 @@ func (f *Field) setLength() {
 
 func (f *Field) transform(subfield bool, ftypes map[string]*Type, handleHRSTQuirk bool, logger *log.Logger) (skip bool, err error) {
 	if f.data[mEXAMPLE] == "" || f.data[mEXAMPLE] == "0" {
-		if f.data[mFNAME] == "heart_rate_source_type" && handleHRSTQuirk {
+		_, ok := extraProductFields[f.data[mFNAME]]
```

Comment:
Can you elaborate more on why these changes were necessary? Do the ebike fields suffer from the same problem as the HRST field in the FIT SDK spreadsheet? Also, you're no longer checking the `handleHRSTQuirk` flag, which might be fine assuming it's always true.

Reply:
These were optional and are not present in the RecordMsg in service so I could not properly access them.  The `handleHRSTQuirk` flag seemed pointless since I always had to build using the flag to make it true and it was not very straight forward at first.

Reply:
What do you mean by optional? Basically every field in a `RecordMsg` is optional.

Reply:
I guess more like optionally complied but they were not in the RecordMsg.  We tried first upgrading out FitSDK to the latest but it was still not include and it started removing some fields which is why we took this route.   @dudanian Could probably better explain since he came up with this solution.

Reply:
Optional is definitely not the right word, but it is probably suffering the same issue as the HRST

Reply:
Weird, ok. Guess we'll roll with it.